File: backend/apps/academy/urls/classes.py
"""
DSAT LMS v2 — Classroom stream URLs
Domain: Academy
Description: Membership-scoped class endpoints (mounted at /api/v1/classes/).
            Distinct from /teacher/classes/, which is the teacher's management
            surface — these are reachable by anyone IN the class.
"""

from django.urls import path

from ..views_stream import (
    ClassCommentDetailView,
    ClassCommentListView,
    ClassPostDetailView,
    ClassStreamView,
    MyClassesView,
)

app_name = "classes"

urlpatterns = [
    path("", MyClassesView.as_view(), name="my-classes"),
    path("<uuid:pk>/stream/", ClassStreamView.as_view(), name="class-stream"),
    path("<uuid:pk>/stream/<uuid:post_pk>/", ClassPostDetailView.as_view(), name="class-post"),
    path(
        "<uuid:pk>/stream/<uuid:post_pk>/comments/",
        ClassCommentListView.as_view(),
        name="class-comments",
    ),
    path(
        "<uuid:pk>/stream/<uuid:post_pk>/comments/<uuid:comment_pk>/",
        ClassCommentDetailView.as_view(),
        name="class-comment",
    ),
]

File: frontend/src/lib/api/classes.ts
// ═══════════════════════════════════════
// DSAT LMS v2 — Classes API (membership-scoped)
// Domain: Academy
// Description: The classroom stream, reachable by anyone IN the class. Distinct
//   from lib/api/teacher.ts, which is the teacher's management surface — these
//   endpoints answer to membership, not to role.
// ═══════════════════════════════════════

import { del, get, getPaginated, post } from './client'
import type { ClassPost, ClassComment, MyClass, Pagination } from '@/types'

export interface NewPostPayload {
  body: string
  kind?: 'post' | 'announcement' | 'material'
  isPinned?: boolean
  allowComments?: boolean
  attachmentIds?: string[]
}

export const classesAPI = {
  /** The classes the current user is in. */
  mine: () => get<MyClass[]>('/classes/'),

  stream: (
    classId: string,
    cursor?: string
  ): Promise<{ data: ClassPost[]; pagination?: Pagination }> =>
    getPaginated<ClassPost>(`/classes/${classId}/stream/`, cursor ? { cursor } : undefined),

  /** Staff only — the server refuses a student. */
  createPost: (classId: string, payload: NewPostPayload) =>
    post<ClassPost>(`/classes/${classId}/stream/`, payload),

  removePost: (classId: string, postId: string) =>
    del<void>(`/classes/${classId}/stream/${postId}/`),

  reply: (classId: string, postId: string, body: string) =>
    post<ClassComment>(`/classes/${classId}/stream/${postId}/comments/`, { body }),

  removeReply: (classId: string, postId: string, commentId: string) =>
    del<void>(`/classes/${classId}/stream/${postId}/comments/${commentId}/`),
}
